src/lib.rs

```diff
@@ -346,36 +380,30 @@ impl From<Host> for Target {
 
 impl From<String> for Target {
     fn from(target_str: String) -> Target {
-        let target_host: Host = target_str.as_str().into();
+        let target_host: TargetTriple = target_str.as_str().into();
         target_host.into()
     }
 }
 
 impl Serialize for Target {
     fn serialize<S: Serializer>(&self, serializer: S) -> Result<S::Ok, S::Error> {
-        match self {
-            Target::BuiltIn { triple } => serializer.serialize_str(triple),
-            Target::Custom { triple } => serializer.serialize_str(triple),
-        }
+        serializer.serialize_str(self.triple())
     }
 }
 
-pub fn run() -> Result<ExitStatus> {
-    let target_list = rustc::target_list(Verbosity::Quiet.into())?;
-    let args = cli::parse(&target_list)?;
-
-    if args.version && args.subcommand.is_none() {
-        let commit_info = include_str!(concat!(env!("OUT_DIR"), "/commit-info.txt"));
-        shell::print(
-            format!(
-                concat!("cross ", env!("CARGO_PKG_VERSION"), "{}"),
-                commit_info
-            ),
-            args.msg_info,
-        )?;
+pub fn run(
+    args: Args,
+    target_list: TargetList,
+    msg_info: MessageInfo,
+) -> Result<Option<ExitStatus>> {
+    if args.all.iter().any(|a| a == "--version" || a == "-V") && args.subcommand.is_none() {
+        println!(
```

Comment:
This undoes the `shell::print` changes, although otherwise it's fine. Probably just needs a rebase.